src/engine/renderer/tr_vbo.cpp

```diff
@@ -353,7 +377,15 @@ static void R_CopyVertexData( VBO_t *vbo, byte *outData, vboData_t inData )
 
 			if ( ( vbo->attribBits & ATTR_TEXCOORD ) )
 			{
-				Vector2Copy( inData.st[ v ], ptr[ v ].texcoord );
+				if ( glConfig2.halfFloatVertexAvailable )
+				{
+					Vector2Copy( inData.f16st[ v ], ptr[ v ].f16TexCoords );
+				}
+				else
+				{
+					// Model only supports half float for now.
+					halfToFloat2( inData.f16st[ v ], ptr[ v ].texCoords );
+				}
```

Comment:
Is this also switched around?

Reply:
It is not. As said in the other comment, the output (second argument) is what will be stored for rendering, so it should be in the format supported by the graphics card/driver, while the input (first argument) is currently always half float.